### PR TITLE
Created Shared Access Serializer

### PR DESCRIPTION
**Pull request by alihadi-mazeh**
Created new Shared Access Serializer that only returns name, id, email and avatar

## Description

Added a shared access serializer that returns the necessary information needed to display data on shared access page. Prior to that shared access page had data being passed in that allowed gave rise for the security flaw that used an exploit with the role id

## Screenshots (if appropriate):

**before the change:**
`{
   "data":[
      {
         "id":"9b1fb09f-2bac-4042-86f5-f33a5f22f2f7",
         "name":"Administrator",
         "email":"admin@admin.com",
         "provider":"greenlight",
         "language":"en",
         "avatar":"http://localhost:3000/assets/default-avatar-12ba46e182bedfae9d6de6c3a414a91b85a2383adff1e06ba4261478c380d3e1.png",
         "verified":false,
         "created_at":"Wednesday October 26, 2022",
         "role":{
            "id":"e0424cd3-a973-4bc2-89c9-207abce74002",
            "name":"Administrator",
            "color":"#29548f"
         }
      },
      {
         "id":"c037b993-9c92-495a-a277-b24e965c3154",
         "name":"B",
         "email":"B@B.b",
         "provider":"greenlight",
         "language":"en",
         "avatar":"http://localhost:3000/assets/default-avatar-12ba46e182bedfae9d6de6c3a414a91b85a2383adff1e06ba4261478c380d3e1.png",
         "verified":false,
         "created_at":"Thursday October 27, 2022",
         "role":{
            "id":"e0424cd3-a973-4bc2-89c9-207abce74002",
            "name":"Administrator",
            "color":"#29548f"
         }
      },
      {
         "id":"51c97c90-2ffa-4b17-a4b0-c29b13024c71",
         "name":"B",
         "email":"C@B.b",
         "provider":"greenlight",
         "language":"en",
         "avatar":"http://localhost:3000/assets/default-avatar-12ba46e182bedfae9d6de6c3a414a91b85a2383adff1e06ba4261478c380d3e1.png",
         "verified":false,
         "created_at":"Thursday October 27, 2022",
         "role":{
            "id":"e0424cd3-a973-4bc2-89c9-207abce74002",
            "name":"Administrator",
            "color":"#29548f"
         }
      }
   ]
}`
**after the change:** 
`{
   "data":[
      {
         "name":"Administrator",
         "email":"admin@admin.com",
         "id":"9b1fb09f-2bac-4042-86f5-f33a5f22f2f7",
         "avatar":"http://localhost:3000/assets/default-avatar-12ba46e182bedfae9d6de6c3a414a91b85a2383adff1e06ba4261478c380d3e1.png"
      },
      {
         "name":"B",
         "email":"B@B.b",
         "id":"c037b993-9c92-495a-a277-b24e965c3154",
         "avatar":"http://localhost:3000/assets/default-avatar-12ba46e182bedfae9d6de6c3a414a91b85a2383adff1e06ba4261478c380d3e1.png"
      },
      {
         "name":"B",
         "email":"C@B.b",
         "id":"51c97c90-2ffa-4b17-a4b0-c29b13024c71",
         "avatar":"http://localhost:3000/assets/default-avatar-12ba46e182bedfae9d6de6c3a414a91b85a2383adff1e06ba4261478c380d3e1.png"
      }
   ]
}`
shared access is displayed the same 
image

### CODE REVIEW

**Comment by alihadi-mazeh** [comment]
I'm not sure if i should add any rspec tests for the change i did in shared access controller. Let me know if i should

File: app/controllers/api/v1/shared_accesses_controller.rb [hunk 1]
```diff
@@ -44,8 +44,7 @@ def shareable_users
                               .where.not(id: [@room.shared_users.pluck(:id) << @room.user_id])
                               .where(role_id: [role_ids])
                               .search(params[:search])
-
-        render_data data: shareable_users, status: :ok
+        render_data data: shareable_users, status: :ok, serializer: SharedAccessSerializer
```
Comment (hadicheaito1) [inline app/controllers/api/v1/shared_accesses_controller.rb:47]: Small comment - Swap the order of `status` and `seriliazer` params; for consistency throughout the app

Comment (hadicheaito1) [reply app/controllers/api/v1/shared_accesses_controller.rb:47]: Besides that looks good! No need for rspec tests

Comment (farhatahmad) [reply app/controllers/api/v1/shared_accesses_controller.rb:47]: Agreed

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**